### PR TITLE
Create directories when needed for the pyfilesystem export plugins

### PR DESCRIPTION
**Pull request by bgruening**
Uploading files to a directory that did not exist in the remote object store, yielded an error, for example in WebDAV.

```
Traceback (most recent call last):
  File "/home/bag/projects/code/galaxy/.venv/lib/python3.9/site-packages/webdavfs/webdavfs.py", line 260, in getinfo
    response = self.client.execute_request(action='info',
  File "/home/bag/projects/code/galaxy/.venv/lib/python3.9/site-packages/webdav3/client.py", line 222, in execute_request
    raise RemoteResourceNotFound(path=path)
webdav3.exceptions.RemoteResourceNotFound: Remote resource: /ERGAPilot/uha not found

During handling of the above exception, another exception occurred:
```
This change will check if the directory exists and creates it before moving the file.

Instead of checking if the directory exists we could also run `fs.makedirs` with `recreate=True`.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:

Setup a WebDAV export and export a file with the name (history name) `/foo/bar/hello_world.txt`.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/files/sources/_pyfilesystem2.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 import logging
 import os
 
+import fs
+
```
Comment (mvdbeek) [inline lib/galaxy/files/sources/_pyfilesystem2.py:7]:
```suggestion
```
This seems to be unused and fails the tests.
Was this an overeager vscode auto-import ? 😆

Comment (bgruening) [reply lib/galaxy/files/sources/_pyfilesystem2.py:7]: Mh, I use it here: `dirname = fs.path.dirname(target_path)`, line 50.

Do you think I can use os.path? I was not sure if there is any plugin that has such a strange syntax that we need `fs`.

**Comment by jmchilton** [comment]
All the rest of the pyfilesystem plugins and hence imports are conditionally imported. I've pushed a commit that might work to just add the base package (https://pypi.org/project/fs/) to the project as a required dependency.

**Comment by bgruening** [comment]
Thanks John!!!

**Pull request merged**